### ISSUE TITLE
add a desktop application that installs the latest geco software on your connected datagrid servers

### ISSUE THREAD

**Issue by stefco** [stefco/geco_vm#67]
basically just does the same thing as you would want to do on the vm, making them into feasible dev machines (but also linking the scripts so that they are useful tools).
- [x] `geco-soft-link` links to the dev repos.
- [x] `geco-soft-dev-update` runs a `git clone` for all geco software repos if necessary and attempts a `git pull` if they are already installed.
- [x] write `geco-soft-update-link-gsissh` which will:
  - connect to the given remote and upload the above scripts
  - run them
  - delete them
  - make sure there is a line in `.bashrc` adding `~/bin` to the path on the remote machine
  - `chmod +x ~/bin`
- [x] write desktop launchers for each that upload these scripts and then execute them on each server.
- [x] write all hosts as right-click options to the desktop launcher
- [x] put these in favorites.

**Comment by stefco** [comment]
http://askubuntu.com/questions/13758/how-can-i-edit-create-new-launcher-items-in-unity-by-hand

**Comment by stefco** [comment]
``` bash
app="${HOME}"/http/home-directory/.medm_launcher/update-and-ssh-data-grid.desktop
cmd="Exec=bash -c 'geco-soft-dev-update-remote ""${host}"" | /usr/bin/zenity --progress --title=\"Updating GECo Software Before Connecting...\" --text=\"Updating software on ${host}...\" --percentage=0 --auto-close --auto-kill; gnome-terminal -x gsissh ${host} & disown'"

# The name of each action is just the host URL:
echo "Actions="$(sed 's/ /;/g' <<<"${hosts[@]}") >> "${app}"

# create an action for each of these host URLs
for host in ${hosts[*]}; do
    echo >> "${app}"
    echo "[Desktop Action ${host}]"  >> "${app}"
    echo "Name=Update GECo Software and Connect with SSH on ${host}"  >> "${app}"
    echo "${cmd}" >> "${app}"
done
```

**Comment by stefco** [comment]
implemented in 9b59c96
